content/blog/2020-07-31-spreadsheet-workflows-using-r/index.Rmarkdown

```diff
@@ -0,0 +1,185 @@
+
+To do that we'll assume the role of a data analyst in two hypothetical scenarios and walkthrough the tidyverse equivalent of some common analyes and tasks one is likely to come across when working in spreadsheets.   
+
+
+For our first story, its day one as a new data analyst and our boss has asked us to take a look at the company's finances.
+They send you a spreadsheet and ask you to analyze spending patterns over the current year and give your assessment into where spending is going overbaord.
+
+We've decide to run our analysis in RStudio to take advatnage of data wrangling and visualization functions from the tidyverse, as well as create a reproducible workflow in case we want to revisit this analysis later.  First let's load in our data and take a look at it in RStudio 
+```{r,message=FALSE}
+#Packages
+library(tidyverse)
+library(readxl)
+
+#Read in and look at our data
+spending <- read_excel("data/spending.xlsx")
+spending
+```
+
+We take note of the month column as well as amount spent, budget, and, expense type. We'll hold off on looking at the column `num_of_expenses`, it may prove useful down the line but not pressing for a quick look into the data.
+
+
+## Nested Logic
+One advantage of spreadsheets is we can use formulas to dictate what should be happening in a specific cell and in a broader application of this, apply them to an entire column.  When creating a column that requires multiple or 'nested' logical conditions, we can take advantage of the `case_when()` function from dplyr.  Similar to a CASE statment in SQL we can set parameters for what a value should be based off conditions from other rows or columns in our data.
+
+Back to our spending data, while data for all 12 months is useful for our summary we want to take a look at spending by quarter, something not provided in the initial dataset.  
+To do this we can create a new column in our dataset using the `mutate()` function and apply `case_when()` within that call.  
+```{r}
+spending_quarter <- spending %>%
+  mutate(quarter = case_when(month %in% c("Jan", "Feb", "Mar") ~ "Q1",
+                             month %in% c("Apr", "May", "Jun") ~ "Q2",
+                             month %in% c("Jul", "Aug", "Sep") ~ "Q3",
+                             month %in% c("Oct", "Nov", "Dec") ~ "Q4",
+                             TRUE ~ "NA"))
+
+spending_quarter
+```
+
+Excellent! We've got our new quarter column set up. Here we specified which months make up each quarter and for good mweasure added an NA condition in the event we've missed something. We can quality check our case_when by examining our data and seeing if any NA values show up which would indicate if a month is spelled differently or if there's an error in our code.  
```

Comment:
How you considered literally juxtaposing this against what this analyst would do if they had to do the task in Excel? I.e. show that nested `=IF()` formula?

I think that would be a really powerful addition.

Reply:
I really like this idea. Will drum up the analysis in excel then add the comparisons in as screenshots